src/python/grpcio_tests/tests_aio/unit/channel_argument_test.py

```diff
@@ -0,0 +1,168 @@
+    (_DISABLE_REUSE_PORT, ((_SOCKET_OPT_SO_REUSEPORT, 0),)),
+)
+
+_NUM_SERVER_CREATED = 100
+
+_GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH = 'grpc.max_receive_message_length'
+_MAX_MESSAGE_LENGTH = 1024
+
+
+class _TestPointerWrapper(object):
+
+    def __int__(self):
+        return 123456
+
+
+_TEST_CHANNEL_ARGS = (
+    ('arg1', b'bytes_val'),
+    ('arg2', 'str_val'),
+    ('arg3', 1),
+    (b'arg4', 'str_val'),
+    ('arg6', _TestPointerWrapper()),
+)
+
+_INVALID_TEST_CHANNEL_ARGS = [
+    {
+        'foo': 'bar'
+    },
+    (('key',),),
+    'str',
+]
+
+
+async def test_if_reuse_port_enabled(server: aio.Server):
+    port = server.add_insecure_port('127.0.0.1:0')
+    await server.start()
+
+    try:
+        another_socket = socket.socket(family=socket.AF_INET)
+        another_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        another_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        another_socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, True)
+        another_socket.bind(('127.0.0.1', port))
+    except OSError as e:
+        assert 'Address already in use' in str(e)
+        return False
+    else:
+        return True
+    finally:
+        another_socket.close()
+
+
+class TestChannelArgument(AioTestBase):
+
+    async def setUp(self):
+        random.seed(_RANDOM_SEED)
+
+    @unittest.skip('https://github.com/grpc/grpc/issues/20667')
+    async def test_server_so_reuse_port_is_set_properly(self):
+
+        async def test_body():
+            fact, options = random.choice(_OPTIONS)
+            server = aio.server(options=options)
+            try:
+                result = await test_if_reuse_port_enabled(server)
+                if fact == _ENABLE_REUSE_PORT and not result:
+                    self.fail(
+                        'Enabled reuse port in options, but not observed in socket'
+                    )
+                elif fact == _DISABLE_REUSE_PORT and result:
+                    self.fail(
+                        'Disabled reuse port in options, but observed in socket'
+                    )
+            finally:
+                await server.stop(None)
+
+        # Creating a lot of servers concurrently
+        await asyncio.gather(*(test_body() for _ in range(_NUM_SERVER_CREATED)))
+
+    async def test_client(self):
+        aio.insecure_channel('[::]:0', options=_TEST_CHANNEL_ARGS)
+
+    async def test_server(self):
+        aio.server(options=_TEST_CHANNEL_ARGS)
+
+    async def test_invalid_client_args(self):
+        for invalid_arg in _INVALID_TEST_CHANNEL_ARGS:
+            self.assertRaises((ValueError, TypeError),
+                              aio.insecure_channel,
+                              '[::]:0',
+                              options=invalid_arg)
+
+    async def test_max_message_length_applied(self):
+        address, server = await start_test_server()
+
+        async with aio.insecure_channel(
```

Comment:
LocalCredentials?

Reply:
Credentials are not supported yet: https://github.com/grpc/grpc/pull/21351